### PR TITLE
Conversions: Improve triggering

### LINKED ISSUE #4112: Conversions: Open up triggering
## Problem

From the anonymized query logs, it seems that the triggering of this IA could be opened up. Below is a dump of these queries to help the contributor improve this instant answer.

### Solved 
Solved by: https://github.com/duckduckgo/zeroclickinfo-goodies/pull/4118
```
1 cm equals how many inches
19 cm equals in
30 cm equals how many inches
180 cm inches conversion
kg to lb conversion
160 lb is how many kg
190 lb = ? kg
35 lb = how many kilo
4 tbsp equals how many cups
1 tbsp = ? tsp
1 gallon equals how many ounces
convert from feet to mile
convert 330 centigrade to Farenheit
```

**Unanswered Queries**
```
167 cm to feet inch
188 cm in feet and inches
177.5 cm to ft in
28551 lb IN KG
1 lb 3 oz
oz to tablespoon conversion
convert tbsp to grams
how many tbsp in 2/3 cup
how many tbsp in an ounce
cubic foot to gallon conversion
convert pounds to newtons
101 degrees in celsius
80 degrees to celsius
65 degrees celsius equals
450 degrees to celsius
minus 40 degrees c to f
350 degrees in centigrade
```

## Solution

1. Add a selection of these queries to [Conversions.t](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/t/Conversions.t)

2. Update the triggering in [Conversions.pm](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/lib/DDG/Goodie/Conversions.pm) to make the tests pass

3. Send your PR

## People to notify

@mintsoft @moollaza 

## Get Started
- [ ] 1) Claim this issue by commenting below
- [ ] 2) Review our [Contributing Guide](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/CONTRIBUTING.md)
- [ ] 3) [Set up your development environment](https://docs.duckduckhack.com/welcome/setup-dev-environment.html), and fork this repository
- [ ] 4) Create a Pull Request

## Resources
- Join [DuckDuckHack Slack](https://quackslack.herokuapp.com/) to ask questions
- Join the [DuckDuckHack Forum](https://forum.duckduckhack.com/) to discuss project planning and Instant Answer metrics
- Read the [DuckDuckHack Documentation](https://docs.duckduckhack.com/) for technical help

---

Instant Answer Page: https://duck.co/ia/view/conversions

### PR DESCRIPTION
**Pull request by adityatandon007**
Conversions: This PR fixes some of the triggers mentioned in #4112 

## Description of new Instant Answer, or changes
This PR resolves some of the mentioned triggering issues for conversions IA

## Related Issues and Discussions

Fixes #4112 

## People to notify
@pjhampton @mintsoft 

## Testing & Review
To be completed by Community Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [ ] Title follows correct format (Specifies Instant Answer + Purpose)
- [ ] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - The **Title** is appropriately named and formatted
    - The **IA topics** are present and appropriate
    - The **Description** is clear and coherent
    - **Source Name** exists if applicable
    - All **Example Queries** trigger on [Beta](https://beta.duckduckgo.com/)

**Code**
- [ ] Adheres to the [DuckDuckGo Style Guide](https://docs.duckduckhack.com/resources/code-style-guide.html)
- [ ] Behaviour is appropriately tested. If improvement, tests are adequately extended.
- [ ] There is no unnecessary files in place (such as editor config files)
- [ ] There is no API keys / secrets present
- [ ] Tests are passing (run `$ duckpan test <goodie_id>`)
    - Tester should report any failures

**Ready to merge?**

- [ ] Has this IA been deployed to and tested on [beta.duckduckgo.com](https://beta.duckduckgo.com/)?
- [ ] For larger commits, has this been approved be more than one community member?
- [ ] The number of reviews is appropriate for this type of PR
- [ ] The commit message is clear, coherent and fitting

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/conversions

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: t/Conversions.t [hunk 1]
```diff
@@ -3226,32 +3226,116 @@ ddg_goodie_test(
...
+            markup_input => '190',
+            raw_input => '190',
+            from_unit => 'pounds',
+            styled_output => '86.183',
+            raw_answer => '86.183',
+            to_unit => 'kilograms',
+            physical_quantity => 'mass'
+        })
+    ),
+    '4 tbsp equals how many cups' => test_zci(
```
Comment (mintsoft) [inline t/Conversions.t]: Can you add a test for `'4 tbsp equals how many cups?'`

File: t/Conversions.t [hunk 2]
```diff
@@ -3226,32 +3226,116 @@ ddg_goodie_test(
...
+            markup_input => '160',
+            raw_input => '160',
+            from_unit => 'pounds',
+            styled_output => '72.575',
+            raw_answer => '72.575',
+            to_unit => 'kilograms',
+            physical_quantity => 'mass'
+        })
+    ),
+    '190 lb = ? kg' => test_zci(
```
Comment (mintsoft) [inline t/Conversions.t]: Can you add a test for some variability in whitespace here? I'm thinking:
* `190lb=?kg`
* `190 lb = ?kg`
* `190 lb =?kg`
* `190lb = ?kg`

**Comment by mintsoft** [comment]
@adityatandon007 Looks good, I'd just like to tighten up the test coverage and then I think it's good to go

File: t/Conversions.t (hunk 2 above)
Comment (adityatandon007) [inline t/Conversions.t]: whether for these queries IA should trigger or not?

File: t/Conversions.t (hunk 1 above)
Comment (adityatandon007) [inline t/Conversions.t]: test added for the above query and the IA triggers for the query

**Comment by adityatandon007** [comment]
@mintsoft I have added an alias for Fahrenheit  bu for `convert pounds to newtons` we already have an entry in yaml file, but the query should be typed as `convert pounds force to newtons`  so as to distinguish it from the `pounds(mass)` so I don't think we need to hack around for this.

screenshot from 2017-04-24 11-02-23

File: t/Conversions.t (hunk 2 above)
Comment (moollaza) [inline t/Conversions.t]: @adityatandon007 all of those should trigger and return a result

Comment (moollaza) [inline t/Conversions.t]: This is live on Beta. Looks like only the format `190lb = ? kg` would trigger.

@adityatandon007 let's make this a little more flexible in another PR. This format should be good to start with 👍

File: share/goodie/conversions/ratios.yml [hunk 3]
```diff
@@ -1320,6 +1320,7 @@ plural: poundals
 ---
 aliases:
   - f
+  - farenheit
```
Comment (moollaza) [inline share/goodie/conversions/ratios.yml]: this seems unnecessary?
@mintsoft can you confirm?

**Comment by moollaza** [comment]
@adityatandon007 this looks pretty good to me.

@mintsoft if you're happy with this, please merge! Would like to get these changes out asap 👍 

We can relax the triggering in another PR if necessary. Seems like the regex just needs some tweaking.

File: share/goodie/conversions/ratios.yml (hunk 3 above)
Comment (mintsoft) [inline share/goodie/conversions/ratios.yml]: It's required for one of the queries in the issue, I suggested this : https://github.com/duckduckgo/zeroclickinfo-goodies/issues/4112#issuecomment-296473545

**Comment by mintsoft** [comment]
Thanks @adityatandon007 if you want to create a second PR for making the `?` style queries more flexible then we can tick some of those off the list

**Comment by bsstoner** [comment]
This was released to production 🚀

File: share/goodie/conversions/ratios.yml (hunk 3 above)
Comment (moollaza) [inline share/goodie/conversions/ratios.yml]: Ah, my mistake didn't realize it was a different spelling!
